### ISSUE TITLE
Weapon targeting cleanup

### ISSUE THREAD

**Issue by wzdev-ci** [Warzone2100/old-trac-import#3041]
`keyword_weapon_targeting` `resolution_fixed` `type_patch (an actual patch, not a request for one)`   |    by Per

___

 * Counterbattery sensors now use real range, and work for allies as well.
 * Radar detector targeting for micro-AI does not require line of sight anymore.
 * actionInsideMinRange() was misunderstood to mean actionInsideMaxRangeAndAllIsFine, while it really
   was checking isActionInsideMinimumRangeAndShouldNotBeAllowed; minimum range is now actually checked!
 * note that micro-AI does not handle minimum range; it never did
 * Droid micro-AI no longer limited to target acquisition within its own vision range,
   can also pick enemies others can see within its fire radius.
 * Micro-AI search radius for structure indirect weapons limited by their own vision range,
   not some random constant.
 * Sensor micro-AI search radius now uses real range, not random contant.
 * A number of duplicate checks in to-hit code removed.
 * To recap: Direct fire weapons can fire at anything within weapons range that the player can see;
   indirect weapons can only fire at anything within the lesser of vision range and weapons range,
   unless they have fire support, in which case vision range is ignored
 * A lot of weapons have a very small minimum range that is totally useless but may produce
   unexpected 'why does my droid not fire' situations where they are _very_ close to target

This patch needs careful review and much testing.

___
_Issue migrated from [trac:3041](https://developer.wz2100.net/ticket/3041) at 2022-04-16 08:46:06 -0700_
